store/batch.go

```diff
@@ -84,7 +85,7 @@ func (b *batch[H]) Append(headers ...H) {
 	b.lk.Lock()
 	defer b.lk.Unlock()
 	for _, h := range headers {
-		b.headers = append(b.headers, h)
+		b.headers = append(b.headers, h) // gap is possible?
```

Comment:
Question to the team: maybe not the best places for a discussion but even before this PR is there a chance to get a gap? `[from, x)` and later `[x+10, to)` by example.